packages/utils/src/modules-sdk/loaders/container-loader-factory.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  Constructor,
+  LoaderOptions,
+  MedusaContainer,
+  ModuleServiceInitializeCustomDataLayerOptions,
+  ModuleServiceInitializeOptions,
+  RepositoryService,
+} from "@medusajs/types"
+import { lowerCaseFirst } from "../../common"
+import { asClass } from "awilix"
+import { abstractServiceFactory } from "../abstract-service-factory"
+import { mikroOrmBaseRepositoryFactory } from "../../dal"
+
+/**
+ * Factory for creating a container loader for a module.
+ *
+ * @param moduleModels
+ * @param moduleServices
+ * @param moduleRepositories
+ */
+export function moduleContainerLoaderFactory({
+  moduleModels,
+  moduleServices,
+  moduleRepositories,
+}: {
+  moduleModels: Record<string, any>
+  moduleServices: Record<string, any>
+  moduleRepositories: Record<string, any>
```

Comment:
would want this to be optional so that if you have no custom repositories they will just be generated for you

Reply:
done